core/domain/exp_domain.py

```diff
@@ -1298,6 +1312,35 @@ def is_demo(self):
         """
         return self.is_demo_exploration_id(self.id)
 
+    @property
+    def android_proto_size_in_bytes(self):
+        """Whether the exploration proto size is up to date.
+
+        Returns:
+            int. Updated exploration's proto size.
+        """
+        if self.android_proto_size_is_stale:
+            self._cached_android_proto_size_in_bytes = self.get_proto_size()
+            self.android_proto_size_is_stale = False
+
+        return self._cached_android_proto_size_in_bytes
+
+    def __setattr__(self, attrname, new_value):
+        """Perform the android_proto_size_is_stale check every time
+        the Exploration class object updates.
+        """
+
+        # If the value of _cached_android_proto_size_in_bytes or
+        # android_proto_size_is_stale gets updated, we don't want to
+        # recompute the exploration's proto size. These both attributes are
+        # the supporting attribute, not included in the proto size calculation.
```

Comment:
These attributes are both supporting attributes which aren't included in the ...

Reply:
Fixed